src/common/components/StatusBar/index.jsx

```diff
@@ -1,11 +1,25 @@
 import React from "react";
-import { StatusBar } from "react-native";
+import { StatusBar, Platform, View } from "react-native";
 import { useSelector } from "react-redux";
 import PropTypes from "prop-types";
 
 const StatusBarComponent = ({ backgroundColor }) => {
   const isStatusBar = useSelector((state) => state.isStatusBar);
   const isNightMode = useSelector((state) => state.isNightMode);
+
```

Comment:
We can extract the common props between android and ios outisde the return statement.

```javascript
// 1 – Compute once per render, not inside JSX
const barStyle = useMemo(
  () => (isNightMode ? 'light-content' : 'dark-content'),
  [isNightMode],
);

// 2 – Props that are common to both platforms
const baseProps = {
  hidden: isStatusBarHidden,
  barStyle,
} as const;

```

Reply:
done